src/inmanta/agent/agent.py

```diff
@@ -914,32 +912,33 @@ async def prepare_resources(
                 if state in const.UNDEPLOYABLE_STATES:
                     undeployable[res_id] = state
             else:
-                if res_type not in logged_resource_types:
-                    logged_resource_types.add(res_type)
-                    logs[res_type].add(
+                if res_type not in failed_resources:
+                    failed_resources[res_type] = (
+                        {res_id},
```

Comment:
Slightly simpler than the if-else is probably to use the `if` only for initialization (to `set()`), and always execute the `add`.

Just a suggestion, feel free to leave as is.